jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/job/Shipment.java

```diff
@@ -541,14 +537,94 @@ public int getPriority() {
 
     @Override
     public Location getStartLocation() {
-        return pickupLocation_;
+        return getPickupActivity().getLocation();
     }
 
 
     @Override
     public Location getEndLocation() {
-        return deliveryLocation_;
+        return getDeliveryActivity().getLocation();
+    }
+
+    public PickupShipmentDEPRECATED getPickupActivity() {
+        return (PickupShipmentDEPRECATED) getActivityList().findByType(PickupShipmentDEPRECATED.NAME)
+                        .get();
+    }
+
+    public DeliverShipmentDEPRECATED getDeliveryActivity() {
+        return (DeliverShipmentDEPRECATED) getActivityList().findByType(DeliverShipmentDEPRECATED.NAME)
+                        .get();
+    }
+
+    // =================== DEPRECATED GETTERS
+
+    @Deprecated
+    public Location getPickupLocation() {
+        return getPickupActivity().getLocation();
+    }
+
+    /**
+     * Returns the pickup service-time.
+     * <p>
+     * <p>
+     * By default service-time is 0.0.
+     *
+     * @return service-time
+     */
+    @Deprecated
+    public double getPickupServiceTime() {
+        return getPickupActivity().getOperationTime();
     }
 
+    @Deprecated
+    public Location getDeliveryLocation() {
+        return getDeliveryActivity().getLocation();
+    }
+
+    /**
+     * Returns service-time of delivery.
+     *
+     * @return service-time of delivery
+     */
+    @Deprecated
+    public double getDeliveryServiceTime() {
+        return getDeliveryActivity().getOperationTime();
+    }
+
+    /**
+     * Returns the time-window of delivery.
+     *
+     * @return time-window of delivery
+     */
+    @Deprecated
+    public TimeWindow getDeliveryTimeWindow() {
+        return getDeliveryTimeWindows().iterator().next();
+    }
+
+    @Deprecated
+    public Collection<TimeWindow> getDeliveryTimeWindows() {
+        return getDeliveryActivity().getTimeWindows();
+    }
+
+    /**
+     * Returns the time-window of pickup.
+     *
+     * @return time-window of pickup
+     */
+    @Deprecated
+    public TimeWindow getPickupTimeWindow() {
+        return getPickupTimeWindows().iterator().next();
+    }
+
+    @Deprecated
+    public Collection<TimeWindow> getPickupTimeWindows() {
+        return getPickupActivity().getTimeWindows();
+    }
+
+    @Override
+    @Deprecated
+    public Capacity getSize() {
+        return getPickupActivity().getSize();
+    }
```

Comment:
Correct. I think .getSize() should not be part of Job.java anymore, but is also part of activity since - this is the great thing here - in can be different over the activites.